packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright IBM Corp. 2022
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+import React, { useState } from 'react';
+import {
+  Button,
+  ButtonSet,
+  Dropdown,
+  TextInput,
+} from 'carbon-components-react';
+import PropTypes from 'prop-types';
+import { Filter16 } from '@carbon/icons-react';
+import { pkg } from '../../settings';
+const componentName = 'AddSelectFilter';
+
+export let AddSelectFilter = ({
+  filterOpts,
+  handleFilter,
+  handleSearch,
+  iconDescription,
+  inputPlaceholder,
+  multi,
+  placeholder,
+  primaryButtonText,
+  searchTerm,
+  secondaryButtonText,
+  textInputLabel,
+}) => {
+  const [filters, setFilters] = useState({});
+  const [open, setOpen] = useState(false);
+
+  const blockClass = `${pkg.prefix}--add-select__global-filter`;
+
+  const searchHandler = (e) => {
+    handleSearch(e.target.value);
+  };
+
+  const onchangeHandler = ({ selectedItem }, id) => {
+    setFilters({
+      ...filters,
+      [id]: selectedItem,
+    });
+  };
+
+  const applyFilters = () => {
+    handleFilter(filters);
+  };
+
+  const resetFilters = () => {
+    setFilters({});
+  };
+
+  const getSelectedItem = (id) => {
+    return filters[id] || '';
+  };
+
+  const showFilter = multi && filterOpts?.length > 0;
+
+  return (
+    <div className={`${blockClass}-search`}>
+      <TextInput
+        id="temp-id"
+        labelText={textInputLabel}
+        placeholder={inputPlaceholder}
+        value={searchTerm}
+        onChange={searchHandler}
+      />
+      {showFilter && (
+        <Button
+          renderIcon={Filter16}
+          hasIconOnly
+          kind="ghost"
+          onClick={() => setOpen(!open)}
+          iconDescription={iconDescription}
+          className={`${blockClass}-toggle`}
+          size="md"
+        />
+      )}
+      {open && (
```

Comment:
I think you should add the `useClickOutside` hook that we have here. I feel like the expected behavior here is that if you click outside the filter panel it should close.